test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -2017,4 +2028,39 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthFo
       });
     });
   });
-});
+  
+  Expect.describe('Registration Flow', function () {
+    itp('does not show the registration button if settings.registration.click is not set', function () {
+      return setup().then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration is empty object', function () {
+      var registration =  {};
+      return setupRegistrationButton(registration.click, registration.label, registration.text).then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration properties are undefined', function () {
```

Comment:
These first 3 tests all look the same to me - it should be one test, i.e. "does not show show the registration link if registration.click is not defined".

There is a missing test here - what happens if registration.click is not a function? Seems like that should also be a no-op.